distributed/protocol/serialize.py

```diff
@@ -445,27 +444,32 @@ def extract_serialize(x):
     >>> extract_serialize(msg)
     ({'op': 'update'}, {('data',): <Serialize: 123>}, set())
     """
-    x2 = type(x)()
-    ser = {}
-    bytestrings = set()
-    _extract_serialize(x, x2, ser, bytestrings)
-    return x2, ser, bytestrings
-
-
-def _extract_serialize(x, x2, ser, bytestrings, path=()):
     typ_x = type(x)
     if typ_x is dict:
         x_items = x.items()
+        x2 = dict()
     elif typ_x is list:
         x_items = enumerate(x)
-        x2.extend(repeat(None, len(x)))
+        x2 = len(x) * [None]
```

Comment:
Also it is worth noting that this change of `list` construction is significantly faster on its own. By reducing the `type` checks performed, we are able to take advantage of this improvement by creating the `list` as we intend without filling up an existing `list` later.

```python
In [1]: %%timeit l = []
   ...: l.extend(repeat(None, 100))
   ...: 
   ...: 
641 ns ± 7.04 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [2]: %%timeit l = []
   ...: l = 100 * [None]
   ...: 
   ...: 
348 ns ± 10.5 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```